cibuildwheel/options.py

```diff
@@ -392,12 +395,22 @@ def globals(self) -> GlobalOptions:
         archs_config_str = args.archs or self.reader.get("archs", sep=" ")
         architectures = Architecture.parse_config(archs_config_str, platform=self.platform)
 
+        container_engine_str = self.reader.get("container-engine")
```

Comment:
This is technically general or linux only, but we don't limit manylinux either, so that's fine. If we ever to limit win/macOS to not allow manylinux, we could also enforce that for container-engine too.

Reply:
There are podman builds for windows, so I think container-engine should be considered a general setting. This could extend to other engines like lima, lxd, or something else that doesn't exist yet.

Reply:
We don't support using a container engine for a Windows target, and I don't really see what that would be useful (you could just run cibuildwheel _in_ the Windows container, after all), and it would require major changes. But this doesn't need any changes and we might not add that level of checking (it's not an easy mistake to make).